velox/type/Timestamp.cpp

```diff
@@ -278,16 +282,31 @@ std::string Timestamp::tmToString(
   appendSmallInt(tmValue.tm_min, out);
   out += ':';
   appendSmallInt(tmValue.tm_sec, out);
-  out += '.';
-  int offset = out.size();
   if (options.precision == TimestampToStringOptions::Precision::kMilliseconds) {
     nanos /= 1'000'000;
+  } else if (
+      options.precision == TimestampToStringOptions::Precision::kMicroseconds) {
+    nanos /= 1'000;
   }
+  if (options.skipZeroSuffix && nanos == 0) {
+    return out;
+  }
+  out += '.';
+  int offset = out.size();
+  int skipZeros = 0;
   while (nanos > 0) {
```

Comment:
Does the `nanos` might be a big value? we could put the while loop in if like that.
if (!options.skipZeroSuffix) {
    while (nanos > 0) {
    out += '0' + nanos % 10;
}
      } else {
 while (nanos > 0) {
      if (nanos % 10 == 0) {
           skipZeros += 1;
      }    
      }
}

Reply:
Updated. Thanks.